client/src/mapper-consents/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { apiErrorMessage, getMapperConsents, useApi } from '../api';
+import { BeatmapInline } from '../BeatmapInline';
+import { IBeatmapset, IMapperBeatmapsetConsent, IMapperConsent } from '../interfaces';
+import { UserInline } from '../UserInline';
+
+interface IMapperBeatmapsetConsentGrouped extends Omit<IMapperBeatmapsetConsent, 'beatmapset_id'> {
+  beatmapset: IBeatmapset
+}
+
+interface IMapperConsentGrouped extends Omit<IMapperConsent, 'beatmapset_consent'> {
+  beatmapset_consents: IMapperBeatmapsetConsentGrouped[]
+}
+
```

Comment:
really don't like doing this but I couldn't find a better way given the limitations of the queries. Ideally when we `LEFT JOIN mapper_consent_beatmapsets` in the query below, we don't get 5 results in the api (which is what currently happens), but rather one result with a list of the beatmapset consents as an attribute. 

To work around this I take said 5 entries and manually consolidate them into a single entry with a list with 5 elements, which is we we need `IMapperConsentGrouped`.

Similarly, `IMapperBeatmapsetConsentGrouped` is necessary because I couldn't find a way to nest an entire `beatmapset` object inside each `beatmapset_consent` given our db query, so I put it at the top level instead (as the `beatmapset_consent_beatmapset` attribute) and then manually set `beatmapset_consent.beatmapset` below, which is why we need a new interface.

let me know if you have a better idea for tackling these issues because they aren't localized to just this one area and will crop up again elsewhere

Reply:
> in the query below, we don't get 5 results in the api (which is what currently happens), but rather one result with a list of the beatmapset consents as an attribute.

yeah in SQL the only result type you can get is a list of rows that all have the same columns.

use one query to load `mapper_consents` + `mapper` + `updater` , and a second query for `mapper_consent_beatmapsets` + `beatmapsets`. then in javascript you can stitch them together however it makes sense to. I haven't reviewed the rest of this yet but ur probably going to end up doing a similar thing as `nominatorsByNominationId` in `/nominations`

Reply:
I moved the javascript logic of stitching together the sql into router.js which made the interfaces a lot nicer, tried for a while to make the query itself nicer as you suggested but couldn't really get it to work, might try again later

Reply:
see 3a62361d5c068c71cb221545f22a5222fbd4127a -- made a little simpler by querying the beatmapset consents separately and attaching them to the consents afterward
this also preserves the ordering you requested in query (which got kinda reset by putting everything into a new map object)

the interfaces are indeed a lot nicer. all I changed was removing `updater` stuff since it wasn't used yet